Tests/RegoTests/BuiltinTests/StringsTests.swift

```diff
@@ -1020,6 +1020,76 @@ extension BuiltinTests.StringsTests {
         ),
     ]
 
+    static let interpolationTests: [BuiltinTests.TestCase] = [
+        BuiltinTests.TestCase(
+            description: "empty",
+            name: "internal.template_string",
+            args: [.array([])],
+            expected: .success("")
+        ),
+        BuiltinTests.TestCase(
+            description: "undefined optional",
+            name: "internal.template_string",
+            args: [.array([.set([])])],
+            expected: .success("<undefined>")
+        ),
+        BuiltinTests.TestCase(
+            description: "primitives",
+            name: "internal.template_string",
+            args: [.array(["foo ", 42, " ", 4.2, " ", false, " ", .null])],
+            expected: .success("foo 42 4.2 false null")
+        ),
+        BuiltinTests.TestCase(
+            description: "primitives, optional",
+            name: "internal.template_string",
+            args: [.array([.set(["foo"]), " ", .set([42]), " ", .set([4.2]), " ", .set([false]), " ", .set([.null])])],
+            expected: .success("foo 42 4.2 false null")
+        ),
+        BuiltinTests.TestCase(
+            description: "collections, optional",
+            name: "internal.template_string",
+            args: [.array([
+                .set([.array([])]), " ",
+                .set([.array(["a", "b"])]), " ",
+                .set([.set([])]), " ",
+                .set([.set(["c"])]), " ",
+                .set([.object([:])]), " ",
+                .set([.object(["d": "e"])]),
+            ])],
+            expected: .success("""
+                [] ["a", "b"] set() {"c"} {} {"d": "e"}
+                """)
+        ),
+        BuiltinTests.TestCase(
+            description: "nested set",
+            name: "internal.template_string",
+            args: [.array([
+                .set([.set([.set([])])]),
+            ])],
+            expected: .success("{set()}")
+        ),
```

Comment:
Some tests are failing right now since `RegoValue` stringification passes through JSON marshaling, which makes e.g. set values not match OPA's string representation. E.g. `[]` and `["a","b"]` instead of `set()` and `{"a", "b"}`.